### PR TITLE
enable config each collection's rate limit by collection properties

### PR DESCRIPTION
**Pull request by weiliu1031**
Signed-off-by: Wei Liu <wei.liu@zilliz.com>
issue: #24008

### CODE REVIEW

File: internal/rootcoord/quota_center.go [hunk 1]
```diff
@@ -394,10 +396,12 @@ func (q *QuotaCenter) calculateReadRates() {
 					zap.Int64("collectionID", collection),
 					zap.Any("queryRate", q.currentRates[collection][internalpb.RateType_DQLQuery]))
 			}
+
+			collectionProps := q.getCollectionLimitConfig(collection)
+			q.guaranteeMinRate(getCollectionSearchRateMin(collectionProps), internalpb.RateType_DQLSearch, collection)
```
Comment (bigsheeper) [inline internal/rootcoord/quota_center.go:401]: maybe we should disable config collection min rate

File: internal/rootcoord/util.go [hunk 2]
```diff
@@ -135,3 +136,220 @@ func getTravelTs(req TimeTravelRequest) Timestamp {
 func isMaxTs(ts Timestamp) bool {
 	return ts == typeutil.MaxTimestamp
 }
+
+func getCollectionInsertRateMax(properties map[string]string) float64 {
```
Comment (bigsheeper) [inline internal/rootcoord/util.go:140]: use switch case with rate type?

File: internal/rootcoord/quota_center.go [hunk 3]
```diff
@@ -403,10 +406,14 @@ func (q *QuotaCenter) calculateReadRates() {
...
 			}
+
+			collectionProps := q.getCollectionLimitConfig(collection)
+			q.guaranteeMinRate(getCollectionRateLimitConfig(collectionProps, common.CollectionSearchRateMinKey), internalpb.RateType_DQLSearch, collection)
+			q.guaranteeMinRate(getCollectionRateLimitConfig(collectionProps, common.CollectionQueryRateMinKey), internalpb.RateType_DQLQuery, collection)
 		}
 
-		q.guaranteeMinRate(Params.QuotaConfig.DQLMinSearchRatePerCollection.GetAsFloat(), internalpb.RateType_DQLSearch, collections...)
-		q.guaranteeMinRate(Params.QuotaConfig.DQLMinQueryRatePerCollection.GetAsFloat(), internalpb.RateType_DQLQuery, collections...)
+		log.Info("QueryNodeMetrics when cool-off",
```
Comment (bigsheeper) [inline internal/rootcoord/quota_center.go:415]: This log will be quite frequently. Suggest to remove it.

File: internal/rootcoord/quota_center.go [hunk 4]
```diff
@@ -700,23 +712,39 @@ func (q *QuotaCenter) resetCurrentRate(rt internalpb.RateType, collection int64)
...
 	}
 	if q.currentRates[collection][rt] < 0 {
 		q.currentRates[collection][rt] = Inf // no limit
 	}
 }
 
+func (q *QuotaCenter) getCollectionLimitConfig(collection int64) map[string]string {
+	collectionInfo, err := q.meta.GetCollectionByID(context.TODO(), collection, typeutil.MaxTimestamp, false)
+	if err != nil {
+		return make(map[string]string)
```
Comment (bigsheeper) [inline internal/rootcoord/quota_center.go:737]: add a rated warn log?

**Comment by bigsheeper** [comment]
/lgtm 

lgtm if log updated

**Review by jiaoew1991 — APPROVED** [review]

**Pull request merged**